src/herder/TxSetFrame.cpp

```diff
@@ -283,7 +349,7 @@ TxSetFrame::makeFromHistoryTransactions(Hash const& previousLedgerHash,
     // This may cause leaks in case of exceptions, so keep the constructors
     // simple and exception-safe.
     return std::shared_ptr<TxSetFrame>(
-        new TxSetFrame(false, previousLedgerHash, txs));
+        new TxSetFrame(false, previousLedgerHash, TxSetFrame::TxPhases{txs}));
```

Comment:
This looks suspicious - wouldn't history contain 2 phases after protocol upgrade?

Reply:
This logic is only used for old-style transaction sets (here when creating TxSetFrame, we pass false for isGeneralized). With the protocol upgrade, we're going to switch to GeneralizedTxSets, and for historical purposes we actually don't reconstruct the TxSetFrame object+convert it to XDR, but instead use the encoded tx set stored in the DB, so it should deal with phases correctly. Please let me know if this makes sense.

Reply:
Ah, right, I just forgot what this is used for (this doesn't have components either).